src/csharp/Microsoft.Spark/Interop/Ipc/JvmThreadPoolGC.cs

```diff
@@ -0,0 +1,152 @@
+
+            // Initialize the GC timer if necessary.
+            if (_activeThreadGCTimer == null)
+            {
+                lock (_activeThreadGCTimerLock)
+                {
+                    if (_activeThreadGCTimer == null && _activeThreads.Count > 0)
+                    {
+                        _activeThreadGCTimer = new Timer(
+                            (state) => GCThreads(),
+                            null,
+                            _threadGCInterval,
+                            _threadGCInterval);
+                    }
+                }
+            }
+
+            return returnValue;
+        }
+
+        /// <summary>
+        /// Try to remove a thread from the pool. If the removal is successful, then the
+        /// corresponding JVM thread will also be disposed.
+        /// </summary>
+        /// <param name="threadId">The ID of the thread to remove.</param>
+        /// <returns>True if success, false if the thread cannot be found.</returns>
+        private bool TryDisposeJvmThread(int threadId)
+        {
+            if (_activeThreads.TryRemove(threadId, out _))
+            {
+                // _activeThreads does not have ownership of the threads on the .NET side. This 
+                // class does not need to call Join() on the .NET Thread. However, this class is
+                // responsible for sending the rmThread command to the JVM to trigger disposal
+                // of the corresponding JVM thread.
+                if ((bool)_jvmBridge.CallStaticJavaMethod("DotnetHandler", "rmThread", threadId))
+                {
+                    _loggerService.LogDebug($"GC'd JVM thread {threadId}.");
+                }
+                else
+                {
+                    _loggerService.LogWarn(
+                        $"rmThread returned false for JVM thread {threadId}. " +
+                        $"Either thread does not exist or has already been GC'd.");
+                    return false;
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Remove any threads that are no longer active.
+        /// </summary>
+        private void GCThreads()
+        {
+            _loggerService.LogDebug("Starting JVM thread GC.");
+            foreach (KeyValuePair<int, Thread> kvp in _activeThreads)
+            {
+                if (!kvp.Value.IsAlive)
+                {
+                    TryDisposeJvmThread(kvp.Key);
+                }
+            }
+
+            lock (_activeThreadGCTimerLock)
+            {
+                // Dispose of the timer if there are no threads to monitor.
+                if (_activeThreadGCTimer != null && _activeThreads.IsEmpty)
+                {
+                    _activeThreadGCTimer.Dispose();
+                    _activeThreadGCTimer = null;
+                }
+            }
+
+            _loggerService.LogDebug("JVM thread GC complete.");
```

Comment:
Can we remove this log and line 130? (We don't have a good logger service, so this will always be printed). I think line 111 should be good enough.

Reply:
Removed.  Also fixed a bug where that function was always returning false.